File: frontend/src/types/questionBank.ts
export type QuestionDifficulty = 'easy' | 'medium' | 'hard';
export type QuestionBankType = 'builtin' | 'custom';

export interface Question {
  id: string;
  question: string;
  expected_answer: string;
  category?: string | null;
  difficulty?: QuestionDifficulty | null;
  source_ref?: string | null;
}

export interface QuestionBank {
  id: string;
  name: string;
  type: QuestionBankType;
  question_count: number;
  categories: string[];
  created_at: string;
}

export interface QuestionBankDetail extends QuestionBank {
  questions: Question[];
}

export interface QuestionBankListResponse {
  items: QuestionBank[];
  total: number;
}

export interface SkippedRow {
  row: number;
  reason: string;
}

export interface ImportQuestionBankResponse {
  bank: QuestionBank;
  imported_count: number;
  skipped_rows: SkippedRow[];
}

export interface QuestionBankListFilters {
  type?: QuestionBankType;
  search?: string;
}

File: backend/app/schemas/question_bank.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

QuestionDifficulty = Literal["easy", "medium", "hard"]
QuestionBankType = Literal["builtin", "custom"]


class QuestionResponse(BaseModel):
    id: str
    question: str
    expected_answer: str
    category: str | None = None
    difficulty: QuestionDifficulty | None = None
    source_ref: str | None = None


class QuestionBankResponse(BaseModel):
    id: str
    name: str
    type: QuestionBankType
    question_count: int
    categories: list[str]
    created_at: datetime


class QuestionBankDetailResponse(QuestionBankResponse):
    questions: list[QuestionResponse]


class QuestionBankListResponse(BaseModel):
    items: list[QuestionBankResponse]
    total: int


class SkippedRow(BaseModel):
    row: int
    reason: str


class ImportQuestionBankResponse(BaseModel):
    bank: QuestionBankResponse
    imported_count: int
    skipped_rows: list[SkippedRow] = Field(default_factory=list)
